### DISCUSSION TITLE
Mix autogenerate and custom order

### DISCUSSION THREAD

**Discussion by lgastler** [withastro/starlight#1609] [Feature Requests]
### What version of `starlight` are you using?

0.21.1

### What is your idea?

We would like to mix the autogeneration of the sidebar with a deeply nested custom order. Since we have some documentation folders which require a more advanced sorting but would like to keep some parent folders of the advanced documentation in `autogeneration` mode.

### Why is this feature necessary?

To enable to manually sort / order a deeper nested documentation folder while allowing others to still be autogenerated. 

### Participation

- [X] I am willing to submit a pull request for this feature.

**Comment by HiDeoo** [comment]
Sharing a plugin based approach I just released that supports defining a [manual order](https://starlight-auto-sidebar.netlify.app/metadata/#order) or change the [sorting order](https://starlight-auto-sidebar.netlify.app/metadata/#sort) for directories and a few other features to tweak autogenerated sidebar groups: [`starlight-auto-sidebar`](https://github.com/HiDeoo/starlight-auto-sidebar)

![starlight-auto-sidebar](https://github.com/user-attachments/assets/8d8e1cda-5a86-44b7-a092-e23e7e8aab3a)

**Reply by lgastler** [reply]
Great, thanks for sharing. Just tried it out definetley helps us out 👍🏼
